Calibration/HcalCalibAlgos/plugins/HcalIsoTrkAnalyzer.cc

```diff
@@ -113,49 +122,59 @@ class HcalIsoTrkAnalyzer : public edm::EDAnalyzer {
-  l3Filter                            = iConfig.getParameter<std::string>("L3Filter");
-  reco::TrackBase::TrackQuality trackQuality_=reco::TrackBase::qualityByName(theTrackQuality);
-  selectionParameters.minPt           = iConfig.getParameter<double>("MinTrackPt");
-  selectionParameters.minQuality      = trackQuality_;
-  selectionParameters.maxDxyPV        = iConfig.getParameter<double>("MaxDxyPV");
-  selectionParameters.maxDzPV         = iConfig.getParameter<double>("MaxDzPV");
-  selectionParameters.maxChi2         = iConfig.getParameter<double>("MaxChi2");
-  selectionParameters.maxDpOverP      = iConfig.getParameter<double>("MaxDpOverP");
-  selectionParameters.minOuterHit     = iConfig.getParameter<int>("MinOuterHit");
-  selectionParameters.minLayerCrossed = iConfig.getParameter<int>("MinLayerCrossed");
-  selectionParameters.maxInMiss       = iConfig.getParameter<int>("MaxInMiss");
-  selectionParameters.maxOutMiss      = iConfig.getParameter<int>("MaxOutMiss");
-  a_coneR                             = iConfig.getParameter<double>("ConeRadius");
-  a_charIsoR                          = a_coneR + 28.9;
-  a_mipR                              = iConfig.getParameter<double>("ConeRadiusMIP");
+  const double isolationRadius(28.9);
+  trigNames_                          = iConfig.getParameter<std::vector<std::string> >("Triggers");
+  theTrackQuality_                    = iConfig.getParameter<std::string>("TrackQuality");
+  processName_                        = iConfig.getParameter<std::string>("ProcessName");
+  l1Filter_                           = iConfig.getParameter<std::string>("L1Filter");
+  l2Filter_                           = iConfig.getParameter<std::string>("L2Filter");
+  l3Filter_                           = iConfig.getParameter<std::string>("L3Filter");
+  reco::TrackBase::TrackQuality trackQuality_=reco::TrackBase::qualityByName(theTrackQuality_);
+  maxRestrictionPt_                   = iConfig.getParameter<double>("MinTrackPt");
+  slopeRestrictionPt_                 = iConfig.getParameter<double>("SlopeTrackPt");
+  selectionParameter_.minPt           = (slopeRestrictionPt_ > 0) ? (maxRestrictionPt_ - 2.5/slopeRestrictionPt_) : maxRestrictionPt_;
+  selectionParameter_.minQuality      = trackQuality_;
+  selectionParameter_.maxDxyPV        = iConfig.getParameter<double>("MaxDxyPV");
+  selectionParameter_.maxDzPV         = iConfig.getParameter<double>("MaxDzPV");
+  selectionParameter_.maxChi2         = iConfig.getParameter<double>("MaxChi2");
+  selectionParameter_.maxDpOverP      = iConfig.getParameter<double>("MaxDpOverP");
+  selectionParameter_.minOuterHit     = iConfig.getParameter<int>("MinOuterHit");
+  selectionParameter_.minLayerCrossed = iConfig.getParameter<int>("MinLayerCrossed");
+  selectionParameter_.maxInMiss       = iConfig.getParameter<int>("MaxInMiss");
+  selectionParameter_.maxOutMiss      = iConfig.getParameter<int>("MaxOutMiss");
```

Comment:
Possible improvement: implement a c'tor for `spr::trackSelectionParameters` getting the values from a `PSet`. Low prority.